### DISCUSSION TITLE
Input improvements

### DISCUSSION THREAD

**Discussion by niikeec** [spacedriveapp/spacedrive#640] [Feature Request]
Opened a discussion related to #633

Initially, I wanted to add the label and error to the FormField, which is a great idea since adding new form field types would be easy. One issue that arises is that to style the field accordingly to the error (red border, ring, ...) you would have to pass the error down to the root Input through context or by getting the field state from the form's context in the form Input component. Labels and errors would also be limited to form fields only. I don't mind if that's a preferred design choice. It has its positives and negatives, would love to hear your opinion.

When it comes to className props, they aren't really required besides the outerClassName,  but you are limiting yourself to one style, now ofc that's not an issue if you know that the style shouldn't/won't change.

**Comment by Brendonovich** [comment]
### Form Fields

Our Form stuff is based on a [blog post of mine](https://www.brendonovich.dev/blog/the-ultimate-form-abstraction), in which I detail adding labels and errors to the wrapper `FormField` component. Yes, this means error and label stuff is only available to form components, but that's kind of the point - it encourages using a form.

### `className` props

This pattern is already used in some places, but I'd like to stop it spreading too much. My preferred solution is to provide JSX inside props, that way we can override more than just `className` and have greater customisation - better to compose components than override them. I'm gonna try and clean a bunch of this up.
